### PR TITLE
move to python-SDK for deploying arm template in configure action

### PR DESCRIPTION
**Pull request by vivishno**
updated arm template parameter file fo support python sdk.
updated main.py

### CODE REVIEW

File: .cloud/.azure/deploy.params.json [hunk 1]
```diff
@@ -1,7 +1,4 @@
 {
```
Comment (ashishonce) [inline .cloud/.azure/deploy.params.json]: I will suggest not to change the basic format of the params file. 
but instead inside the code, read only the parameters section of the file. 

reason is we want to give user a very cli native experiences.

so inside the code just read the parameters part.

File: code/main.py [hunk 2]
```diff
@@ -63,31 +54,52 @@ def main():
...
+    parameters=jsonobject
+    deployment_properties = {
+        'properties':{
+            'mode': DeploymentMode.incremental,
+            'template': template,
+            'parameters': parameters
+        }
+     }
+    try:
+        deployment_async_operation = client.deployments.create_or_update(
```
Comment (ashishonce) [inline code/main.py:92]: there is option to validate the deployment before creating or updating. 
can you check , using that api as well, before we deploy

File: code/main.py [hunk 3]
```diff
@@ -63,31 +54,52 @@ def main():
...
-    if success:
-        print(deploy_functionApp(template_file_file_path, template_params_file_path, resource_group))
+    credentials=None
+    try:
+        credentials = ServicePrincipalCredentials(
+             client_id=service_principal_id,
+             secret=service_principal_password,
+             tenant=tenant_id
+          )
+    except Exception as ex:
```
Comment (ashishonce) [inline code/main.py:66]: Let's also add standard exceptions as added in aml workspace action.  also we can handle the exceptions commonly occuring

File: code/main.py [hunk 4]
```diff
@@ -63,31 +54,52 @@ def main():
...
-        login_result = subprocess.check_output(command, shell=True)
-        print(login_result)
-    except Exception as ex:
-        print(ex)
-        return;
-
+    
     success = False
+    jsonobject = None
     try:
```
Comment (ashishonce) [inline code/main.py]: we can move this part to a utility function

**Review by ashishonce — CHANGES_REQUESTED** [review]

File: code/main.py (hunk 2 above)
Comment (vivishno) [reply code/main.py:92]: added validation

File: code/main.py (hunk 3 above)
Comment (vivishno) [reply code/main.py:66]: added exceptions to required places

File: .cloud/.azure/deploy.params.json (hunk 1 above)
Comment (vivishno) [reply .cloud/.azure/deploy.params.json]: reverted the change and handling this inside code

File: code/main.py (hunk 4 above)
Comment (vivishno) [reply code/main.py]: moved to utils

**Review by ashishonce — APPROVED** [review]

**Pull request merged**